telemetry_package/main.py

```diff
@@ -1,28 +1,73 @@
-import sys
+import json
 import rclpy
-from rclpy.node import Node
+import socket
+import os
+import select
 
+from rcl_interfaces.msg import Log
 from common_package_py.common_node import CommonNode
 
+DEFAULT_UNIX_SOCKET_PATH = "/tmp/thi_drone"
 
 class TelemetryNode(CommonNode):
-    """A class that represents a telemetry node.
-    """
 
-    def __init__(self, id: str):
-        """Creates a new TelemetryNode.
+    def __init__(self):
+        # The creation of the needed sockets takes place in the constructor
+        super().__init__('telemetry_node')
 
-        Args:
-            id (str): Unique node id
-        """
+        if os.path.exists(DEFAULT_UNIX_SOCKET_PATH): 
+            raise FileExistsError('Socket already exists')
 
-        super().__init__(id)
+        try:         
+            # Create server socket
+            self.server_sock = socket.socket(socket.AF_UNIX, socket.        SOCK_STREAM)
+
+            # Bind server socket and listen for a connection. The node is blocked as long as a connection does not exist
+            self.server_sock.bind(DEFAULT_UNIX_SOCKET_PATH)
+
+            print("Blocking until connection is established...")
+            self.server_sock.listen(1)
+
+            # Getting the reference for the client socket
+            readable, _, _, = select.select([self.server_sock], [], [])
+            if self.server_sock in readable:            
+                # Accept the connection
+                self.client_sock, _ = self.server_sock.accept()
+                print('Connection accepted. Creating Subscription')
+
+                
+            # Creating the subscription after the connection has accepted ensures that the callback function is executed only after the socket is connected           
+            self.subscription = self.create_subscription(
+                Log,
+                '/rosout',
+                self.subscription_callback, 
+                100)
+            self.subscription
+
+        # In case of exceptions, close the socket and remove its path from disk
+        except Exception as e:
+            print(f'Error occurred: {e}')
+            self.server_sock.close()
+            os.remove(DEFAULT_UNIX_SOCKET_PATH)
+        
+ 
+    def subscription_callback(self, log_msg):
+        try: 
+            '''
+            Apparently, some characters in the log messages upset something in the json.loads(<received_data>.decode()) function. This is just a brute sanitization of the string, which needs to be implemented in a function of its own
+            '''
+            log_msg.msg = log_msg.msg.replace(':', '').replace('"', '').replace("'", "").replace("\{", "").replace("}", "")
+
+            self.client_sock.sendall(json.dumps({"type": "std" , "content" :  log_msg.msg}).encode())
```

Comment:
```suggestion
            self.client_sock.sendall(json.dumps({"type": "std" , "content" :  log_msg.msg}).encode() + b"x17")
```
@maf0115 this will fix the errors.

@Philipp66904 was right, pointing out, that the socket seems to be the issue, not the JSON. 

[See here](https://github.com/THI-Drone/Ground_Station/blob/dev/test/test_unix_socket/receiver.py), for adapted receiver dummy script.

Reply:
@maf0115 since node-library I use handles ends of readable streams it should be fine without this addition, this would just be for your debugging purposes.
In production please omit the ETB character.